src/app/extensions/order-templates/shared/order-template-widget/order-template-widget.component.ts

```diff
@@ -0,0 +1,54 @@
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+
+import { GenerateLazyComponent } from 'ish-core/utils/module-loader/generate-lazy-component.decorator';
+import { whenTruthy } from 'ish-core/utils/operators';
+
+import { OrderTemplatesFacade } from '../../facades/order-templates.facade';
+import { OrderTemplate } from '../../models/order-template/order-template.model';
+
+@Component({
+  selector: 'ish-order-template-widget',
+  templateUrl: './order-template-widget.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+@GenerateLazyComponent()
+export class OrderTemplateWidgetComponent implements OnInit, OnDestroy {
+  loading$: Observable<boolean>;
+  orderTemplates$: Observable<OrderTemplate[]>;
+
+  dummyProduct = { sku: 'dummy', available: true };
+
+  orderTemplates: OrderTemplate[];
+
+  private destroy$ = new Subject();
+
+  constructor(private facade: OrderTemplatesFacade) {}
+
+  ngOnInit() {
+    this.loading$ = this.facade.orderTemplateLoading$;
+    this.orderTemplates$ = this.facade.orderTemplates$;
+
+    this.orderTemplates$
+      .pipe(
+        whenTruthy(),
+        map(orderTemplates => orderTemplates.slice(0, 3)),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(orderTemplates => {
+        this.orderTemplates = orderTemplates;
+      });
+  }
+
+  addTemplateToBasket(orderTemplateId: string) {
```

Comment:
Why does this method use the orderTemplateId instead of the complete orderTemplate object? You could call addTemplateToBasket(orderTemplate) from the template and remove the "find" logic. Then you would also not have to subscribe and save in a local variable.